cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/PipelineTriggersUtils.java

```diff
@@ -0,0 +1,144 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.cdap.datapipeline.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.schedule.ProgramStatusTriggerInfo;
+import io.cdap.cdap.etl.api.lineage.field.FieldOperation;
+import io.cdap.cdap.etl.common.BasicArguments;
+import io.cdap.cdap.etl.common.FieldOperationTypeAdapter;
+import io.cdap.cdap.etl.proto.v2.ArgumentMapping;
+import io.cdap.cdap.etl.proto.v2.PluginPropertyMapping;
+import io.cdap.cdap.etl.proto.v2.TriggeringPipelineId;
+import io.cdap.cdap.etl.proto.v2.TriggeringPropertyMapping;
+import io.cdap.cdap.internal.io.SchemaTypeAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Handles parsing of pipeline runtime arguments mapping.
+ */
+public final class PipelineTriggersUtils {
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .registerTypeAdapter(FieldOperation.class, new FieldOperationTypeAdapter()).create();
+  private static final String RESOLVED_PLUGIN_PROPERTIES_MAP = "resolved.plugin.properties.map";
+  private static final Type STAGE_PROPERTIES_MAP = new TypeToken<Map<String, Map<String, String>>>() { }.getType();
+  private static final Logger LOG = LoggerFactory.getLogger(PipelineTriggersUtils.class);
+
+  private PipelineTriggersUtils() {
+  }
+
+  /**
+   * Parse the arguments mapping from the pipeline schedule and add to mappings.
+   * If compositeTriggerEnabled is true, the code will iterate through all the argument mappings and only adds
+   * those which match pipeline IDs.
+   * Example: for given TriggeringPropertyMapping as "{"arguments":[{"source":"source1","target":"target1","pipelineId"
+   * :{"namespace":"default","pipelineName":"Pipeline1"}}，{"source":"source2","target":"target2","pipelineId":
+   * {"namespace":"default","pipelineName":"Pipeline2"}}]}" and a trigger {"namespace":"default","pipelineName":
+   * "Pipeline1"} which has runtime arguments "{"source1": "value1","source2":"value2"}". When composite pipeline
+   * trigger is enabled, only {"target1": "value1"} is added. Otherwise, the code will ignore the ID check and
+   * {"target1": "value1", "target2": "value2"} is added.
+   *
+   * @param triggerInfo {@link ProgramStatusTriggerInfo} with plugin properties
+   * @param propertiesMapping  {@link TriggeringPropertyMapping} for using the plugin
+   * @param compositeTriggerEnabled  whether the composite trigger is enabled
+   */
+  public static void addSchedulePropertiesMapping(Map<String, String> mappings,
```

Comment:
Most of the code here is from `SmartWorkflow.java`, with slight difference that now it checks the ID if `compositeTriggerEnabled` is true